admin/broadleaf-open-admin-platform/src/main/java/org/broadleafcommerce/openadmin/web/service/FormBuilderServiceImpl.java

```diff
@@ -1196,7 +1196,7 @@ public EntityForm buildMapForm(MapMetadata mapMd, final MapStructure mapStructur
         } else {
             keyField = new Field().withName("key")
                                 .withFieldType(SupportedFieldType.STRING.toString())
-                                .withFriendlyName("Key");
+                                .withFriendlyName(mapStructure.getKeyPropertyFriendlyName()");
```

Comment:
@khanmark Looks like there's a stray `"` (double quotes) here.  Otherwise review/testing is good.